pkg/nfd-master/nfd-master.go

```diff
@@ -758,21 +797,29 @@ func (m *nfdMaster) nfdAPIUpdateAllNodes() error {
 	}
 
 	for _, node := range nodes.Items {
-		m.nodeUpdaterPool.addNode(node.Name)
+		m.updaterPool.addNode(node.Name)
 	}
 
 	return nil
 }
 
-func (m *nfdMaster) nfdAPIUpdateOneNode(cli k8sclient.Interface, node *corev1.Node) error {
-	if m.nfdController == nil || m.nfdController.featureLister == nil {
-		return nil
+// mergeNodeFeatures merges the features of the given NodeFeature objects into a single NodeFeatureSpec.
+func (m *nfdMaster) mergeNodeFeatures(nodeName string) (*nfdv1alpha1.NodeFeature, error) {
+	nodeFeatures := &nfdv1alpha1.NodeFeature{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: nodeName,
```

Comment:
I think piggy-backing on the Name field is fine here. You could add a comment to the function description ("docstring") that the function returns a NodeFeature object where the name field is set to the node name. WDYT?